rasa/core/training/story_conflict.py

```diff
@@ -304,17 +304,28 @@ def _get_previous_event(
     # A typical state is, for example,
     # `{'prev_action_listen': 1.0, 'intent_greet': 1.0, 'slot_cuisine_0': 1.0}`.
     # We need to look out for `prev_` and `intent_` prefixes in the labels.
-    for turn_label in state:
-        if (
-            turn_label.startswith(PREV_PREFIX)
-            and turn_label.replace(PREV_PREFIX, "") != ACTION_LISTEN_NAME
-        ):
-            # The `prev_...` was an action that was NOT `action_listen`
-            return "action", turn_label.replace(PREV_PREFIX, "")
-        elif turn_label.startswith(INTENT + "_"):
+    for origin, substate in state.items():
+        if origin == PREVIOUS_ACTION:
+            if (
+                "action_name" in substate
+                and substate["action_name"] != ACTION_LISTEN_NAME
+            ):
+                # The `prev_...` was an action that was NOT `action_listen`
+                assert isinstance(substate["action_name"], str)
+                return "action", substate["action_name"]
+            elif "action_text" in substate:
```

Comment:
As you return in the previous line you could change the `elif` to an `if`.

Reply:
Don't you think `elif` makes it easier to read? Then you don't have to care about what happens inside the `if`-block. One of our code-quality checks used to always tell me that I should use `elif` to reduce "cognitive complexity" and I think that makes sense.

Reply:
I think we added something to our coding guidelines that if you do return in one if branch you should not continue with else or elif. But not 100% sure. Do not have a strong opinion here. We can also leave it as it is.